cylc/flow/batch_sys_handlers/slurm.py

```diff
@@ -44,7 +45,7 @@ class SLURMHandler():
     @classmethod
     def format_directives(cls, job_conf):
         """Format the job directives for a job file."""
-        job_file_path = re.sub(r'\$HOME/', '', job_conf['job_file_path'])
+        job_file_path = os.path.expandvars(job_conf['job_file_path'])
```

Comment:
Is this safe on systems where Slurm is accessed via a login node?

Reply:
I think the old code is correct, the idea is not to expand $HOME but to remove it entirely, when slurm sees a relative path it will root it from `~`.

Reply:
On investigation I think that it's correct as it is - I think the master version of this relies on `get_host_item` doing magic before filling in the job_conf['job_file_path']